test/runtests.jl

```diff
@@ -51,6 +52,46 @@ Random.seed!(100)
     @test cost2 ≈ cost
 end
 
+@testset "1D Optimal Transport for Convex Cost" begin
+    # Continuous Case
+    μ = Normal(0, 2)
+    ν = Normal(10, 2)
+    c(x, y) = abs(x - y)
+    γ = optimal_transport_plan(c, μ, ν)
+
+    @test optimal_transport_cost(c, μ, ν) ≈ 10 atol = 1e-5
+    @test optimal_transport_cost(c, μ, ν; plan=γ) ≈ 10 atol = 1e-5
+
+    # Discrete Case
+    n, m = 100, 150
+
+    μ = rand(n)
+    ν = rand(m) .+ 0.5
+    μ_n = rand(n)
+    ν_m = rand(m)
+    μ_n = μ_n / sum(μ_n)
+    ν_m = ν_m / sum(ν_m)
```

Comment:
This avoids some allocations caused by temporary arrays and makes it clearer what these arrays are:
```suggestion
    μ_probs = rand(n)
    ν_probs = rand(m)
    μ_probs ./= sum(μ_probs)
    ν_probs ./= sum(ν_probs)
```